### PR TITLE
Fix Alt+Enter fullscreen in DirectX 12 example.

### PR DESCRIPTION
**Pull request by PathogenDavid**
This PR fixes https://github.com/ocornut/imgui/issues/4346

Not sure why this example was recreating the entire swap chain on resize (it's been like that forever), but `ResizeBuffers` is the correct way to react to `WM_SIZE`.

Only the Direct3D 12 sample was affected, other samples and viewport-aware backends are fine:

* [Direct3D 10 Sample](https://github.com/ocornut/imgui/blob/c881667c00655c98dba41deb942587e0041d0ed0/examples/example_win32_directx10/main.cpp#L229)
* [Direct3D 11 Sample](https://github.com/ocornut/imgui/blob/c881667c00655c98dba41deb942587e0041d0ed0/examples/example_win32_directx11/main.cpp#L233)
* [Direct3D 10 Docking Backend](https://github.com/ocornut/imgui/blob/7bfc379a23e97b6777eb80aafb50a8e6248904cd/backends/imgui_impl_dx10.cpp#L666)
* [Direct3D 11 Docking Backend](https://github.com/ocornut/imgui/blob/7bfc379a23e97b6777eb80aafb50a8e6248904cd/backends/imgui_impl_dx11.cpp#L683)
* [Direct3D 12 Docking Backend](https://github.com/ocornut/imgui/blob/7bfc379a23e97b6777eb80aafb50a8e6248904cd/backends/imgui_impl_dx12.cpp#L997)

### CODE REVIEW

**Comment by PathogenDavid** [comment]
Closing while I investigate a quirk I found.

**Comment by PathogenDavid** [comment]
TL;DR: Had a false alarm on a bug in this change. Found unnecessary device object recreation in the process.

--------



I finished investigating the quirk. While testing this change with the docking branch I noticed a debug layer crash happening in `ImGui_ImplDX12_InvalidateDeviceObjects`.

I did not determine the root cause of the crash (other than that it is specific to the docking backend -- will investigate more later), but it did make me realize I got tunnel vision and did not notice this unnecessary destruction/creation of backend-owned device objects.

Since this PR is already touching this area and it is related to window resizing, I removed those calls as well.
